components/formats-gpl/src/loci/formats/in/MetamorphReader.java

```diff
@@ -871,12 +874,13 @@ else if (firstSeriesChannels != null) {
 
           if ((int) wave[waveIndex] >= 1) {
             // link LightSource to Image
+            int laserIndex = i * getEffectiveSizeC() + c;
```

Comment:
No, I don't think so.  The indexes on the ```setDetectorSettings*``` methods are for ```Image``` and ```Channel```, not ```Instrument```, so I think what is here will continue to work as expected.